artview/components/display.py

```diff
@@ -0,0 +1,297 @@
+        else:
+            print("Overwriting text")
+            select = self.dispCombo.findText(self.choice['text'])
+            try:
+                self.dispChoiceList[select]['instance'].remove()
+            except:
+                print("No instance found")
+            self.choice['instance'] = self.display.ax.text(
+                self.choice['xpos'], self.choice['ypos'],
+                self.choice['text'], style=self.choice['style'],
+                color=self.choice['col'], fontsize=self.choice['size'])
+            self.dispChoiceList[select] = self.choice
+            self.dispCombo.setCurrentIndex(select)
+        # Write the updated text item to the Display Text instance
+        self.display.disp_text[self.choice['text']] = self.choice
+
+        # Redraw the canvas to place new text
+        self.display.fig.canvas.draw()
+
+    def delDispText(self):
+        '''Delete selected Display text boxes.'''
+        self.choice_key = str(self.ent_tex.text())
+        if self.dispCombo.currentText() in self.display.disp_text.keys():
+            del self.display.disp_text[self.choice_key]
+
+        delselect = self.dispCombo.findText(self.choice_key)
+        if delselect != -1:
+            self.dispChoiceList[delselect]['instance'].remove()
+            del self.dispChoiceList[delselect]
+            self.dispCombo.removeItem(delselect)
+            self.display.fig.canvas.draw()
+
+        self.choice = self._blank_entries()
+        self.dispCombo.setCurrentIndex(0)
+        self._rebuild_entry()
+
+    def clrDispText(self):
+        '''Clear all Display text boxes.'''
+        self.display.disp_text = {}
+        self.choice = self._blank_entries()
+        # Delete each entry, need to reverse indices
+        for i in range(self.dispCombo.count())[::-1]:
+            if i > 0:
+                self.dispChoiceList[i]['instance'].remove()
+                del self.dispChoiceList[i]
+                self.dispCombo.removeItem(i)
+        self.display.fig.canvas.draw()
+        self.dispCombo.setCurrentIndex(0)
+        self._rebuild_entry()
+##        self.display._update_plot()##NG
+
+    #####################
+    #   Entry Methods   #
+    #####################
+
+    def _rebuild_entry(self):
+        '''Reset the entries based upon input.'''
+        self.ent_tex.setText(str(self.choice['text']))
+        self.ent_xpos.setText(str(self.choice['xpos']))
+        self.ent_ypos.setText(str(self.choice['ypos']))
+        self.ent_texc.setText(str(self.choice['col']))
+        self.ent_texsz.setText(str(self.choice['size']))
+        self.ent_texst.setText(str(self.choice['style']))
+
+    def _blank_entries(self):
+        '''Initialize Add Text Instance.'''
+        blank_text = {'text': '', 'xpos': '', 'ypos': '',
+                      'col': '', 'size': '', 'style': ''}
+        return blank_text
+
+    def _check_entries(self):
+        '''Check that entries are valid.'''
+        #NG Put in warnings?
```

Comment:
make some kind of graphical warning, text in the terminal is likely to be over seen